client/src/main/java/io/confluent/kafka/schemaregistry/AbstractSchemaProvider.java

```diff
@@ -101,13 +101,35 @@ protected static boolean canLookupIgnoringVersion(
           : current.copy(new Metadata(null, null, null), current.ruleSet());
       ParsedSchema newPrev = prev.copy(
           Metadata.removeConfluentVersion(prev.metadata()), prev.ruleSet());
-      // This handles the case where a schema is sent without confluent:version
+      // This handles the case where current schema is without confluent:version
+      return newSchema.deepEquals(newPrev);
+    } else if (schemaVer != null && prevVer == null) {
+      if (!versionsMatch(schemaVer, prev.version())) {
+        // The incoming confluent:version must match the actual version of the prev schema
```

Comment:
why do we require actual version of the prev schema matches `confluent:version` of the schema being lookedup

Reply:
Other places in the code may assume that the versions match.  If they don't match, it would be confusing to other parts of the code, and would make the code that has to deal with the versions not match more confusing.